Automattic/vip-design-system
# Fix/new dialog references

## Description

- Fixes #86 

- This PR fixes the issue when the Dialog closes the focus doesn't get back to the previous trigger. The problem is that some of our components don't support passing a `ref` attribute. This PR fixes the Button reference and adds support to export the `Close` primitive container so that the VIP Dashboard can implement custom Close buttons inside of the Dialog.

- Radix Dialog upgraded to 1.0.0

## Checklist


## Steps to Test — Focus fix

- Follow the steps on issue #86 

## Steps to Test - Custom Close

- Go to http://localhost:6006/?path=/story/newdialog--custom-close and use the internal close button. The focus should stay on the original trigger.
